src/any_agent/tracing/instrumentation/google.py

```diff
@@ -0,0 +1,215 @@
+from __future__ import annotations
```

Comment:
As mentioned, I'd even instrument this and not use an api that might change at any time.

Reply:
> As mentioned, I'd even instrument this and not use an api that might change at any time.

Not sure I follow what you mean. Is it more likely for a public API (the callbacks) to change than some internal method we would patch?